Clarify FETCH_OK fields (#936)
moq-wg/moq-transport: draft-ietf-moq-transport.md

Ascending (0x1) or Descending (0x2) order by group. See {{priorities}}.
Values of 0x0 and those larger than 0x2 are a protocol error.

* End Of Track: 1 if all Objects have been published on this Track, and
  the End Location is the final Object in the Track, 0 if not.

* End: The largest object covered by the FETCH response.

>   The End value is determined by taking the minimum of two values:

Author: This is still ambiguous whether {group,object + 1} is used in FETCH_OK.  In my implementation, I think it's easiest to use + 1, and I _think_ End.object == 0 is required to correctly denote the end of the group when there is no END_OF_GROUP marker

```
S->P: FETCH start=10,0 end=10,0  // requesting all of group 10
P->S: FETCH_OK, 10,0
objects 10,0 - 10,10             // 10,10 is the last object in the group
```
If FETCH_OK had 10/10 (actual end) it is ambiguous with the case where 10/10 is the Largest object.

0 if not.

* End: The largest object covered by the FETCH response.
  The End value is determined by taking the minimum of two values:
   - The {EndGroup, EndObject} specified in the FETCH request.

>    - The largest known {Group, Object} available.

Author: I realized the description of "End of Track" above is a little off now.  I think it should be 1 if End is the final object in the track.

Should I update it in this PR?

  track, the largest known {group,object} at the relay is used.  For tracks
  with a requested end larger than what is cached without an active
  subscription, the relay makes an upstream request in order to satisfy the

>   FETCH.  If End is smaller than the Start Location in the corresponding

Reviewer: Note: The preceding text for End is quite difficult to read.  I'm not sure if you want to rewrite it now or we should do it in a different PR?

Reviewer 2: +1 , would be nice to fix in this PR if possible as they are related

Ascending (0x1) or Descending (0x2) order by group. See {{priorities}}.
Values of 0x0 and those larger than 0x2 are a protocol error.

* End Of Track: 1 if all Objects have been published on this Track, and
  the End Location is the final Object in the Track, 0 if not.

* End Location: The largest object covered by the FETCH response.

>   The End value is determined by taking the minimum of two values:
>    - The {End Group, End Object} specified in the FETCH request.
>    - The largest known, possibly final, {Group, Object} available
> 

Author: ```suggestion
  The End Location is determined as follows:
   - If End Object in the FETCH request was 0 and the response covers the last
     Object in the Group, End Location is {Fetch.End Group, 0}
   - If the requested FETCH End was beyond the Largest known (possibly final)
     Object, End Location is {Largest.Group, Largest.Object + 1}
   - Otherwise, End Location is {Fetch.End Group, Fetch.End Object} 

```

Ascending (0x1) or Descending (0x2) order by group. See {{priorities}}.
Values of 0x0 and those larger than 0x2 are a protocol error.

* End Of Track: 1 if all Objects have been published on this Track, and
  the End Location is the final Object in the Track, 0 if not.

* End Location: The largest object covered by the FETCH response.
  The End Location is determined as follows:
   - If End Object in the FETCH request was 0 and the response covers the last
     Object in the Group, End Location is {Fetch.End Group, 0}
   - If the requested FETCH End was beyond the Largest known (possibly final)
     Object, End Location is {Largest.Group, Largest.Object + 1}
   - Otherwise, End Location is {Fetch.End Group, Fetch.End Object} 

> 

Author: If the track is over, the final object is 5, 3, and the requested end is 5,0, then this text allows both:

`End=5, 0` and `End of Track=1, End=5, 4` as valid FETCH_OK.  I'd like to make only a single way to encode that, without making the rules too complicated.

Author: I reordered the rules to prefer "End of Track=1, End Location=5,4" interpretation.

  If the relay is subscribed to the track, it uses its knowledge of the largest

>   {Group, Object} to set End.  If if is not subscribed and the requested End

Reviewer: ```suggestion
  {Group, Object} to set End Location.  If if is not subscribed and the requested End
```

Ascending (0x1) or Descending (0x2) order by group. See {{priorities}}.
Values of 0x0 and those larger than 0x2 are a protocol error.

* End Of Track: 1 if all Objects have been published on this Track, and
  the End Location is the final Object in the Track, 0 if not.

* End Location: The largest object covered by the FETCH response.
  The End Location is determined as follows:

>    - If End Object in the FETCH request was 0 and the response covers the last
>      Object in the Group, End Location is {Fetch.End Group, 0}
>    - If the requested FETCH End was beyond the Largest known (possibly final)
>      Object, End Location is {Largest.Group, Largest.Object + 1}
>    - Otherwise, End Location is {Fetch.End Group, Fetch.End Object}

Author: ```suggestion
   - If the requested FETCH End Location was beyond the Largest known (possibly
     final) Object, End Location is {Largest.Group, Largest.Object + 1}
   - If End Location.Object in the FETCH request was 0 and the response covers
     the last Object in the Group, End Location is {Fetch.End Location.Group, 0}
   - Otherwise, End Location is Fetch.End Location
```

The Subgroup ID field of an object with a Forwarding Preference of "Datagram"
(see {{object-properties}}) is set to the Object ID.

> A valid FETCH stream always contains at least one Object.  If a subscriber
> receives a FIN on a FETCH stream without any Objects, it MUST close the session
> with a `Protocol Violation`.
> 

Author: ```suggestion
```
